### PR TITLE
chore: nix flake for notty

### PR DESCRIPTION
**Pull request by Alizter**
Add nix flake and CI job.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        packages = rec {
+          default = notty;
+          notty = buildDunePackage {
+            pname = "notty";
+            version = "n/a";
+            src = ./.;
+            duneVersion = "3";
+            nativeBuildInputs = with pkgs.ocamlPackages; [ cppo ];
+            propagatedBuildInputs = with pkgs.ocamlPackages; [ uutf ];
+            checkInputs = with pkgs.ocamlPackages; [ ppx_inline_test ppx_bench core_bench ppx_expect ];
```
Comment (rgrinberg) [inline flake.nix]: why do we need these test deps?

Comment (Alizter) [reply flake.nix]: Whoops, fixed.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            src = ./.;
+            duneVersion = "3";
+            nativeBuildInputs = with pkgs.ocamlPackages; [ cppo ];
+            propagatedBuildInputs = with pkgs.ocamlPackages; [ uutf ];
+            doCheck = true;
+          };
+        };
+        devShells.default = pkgs.mkShell {
+          inputsFrom = pkgs.lib.attrValues packages;
+          buildInputs = with pkgs.ocamlPackages; [ ocaml-lsp pkgs.ocamlformat ];
```
Comment (rgrinberg) [inline flake.nix]: does this package have an ocamlformat config?

Comment (Alizter) [reply flake.nix]: It does not use ocamlformat AFAICT.

Comment (rgrinberg) [reply flake.nix]: Then this dependency isn't necessary as we can't use ocamlformat either

**Comment by Alizter** [comment]
@rgrinberg The CI will not work until you turn on GitHub actions for this repo.

**Pull request merged**